hermes-consumers/src/main/java/pl/allegro/tech/hermes/consumers/consumer/BatchConsumer.java

```diff
@@ -59,18 +65,23 @@ public BatchConsumer(MessageReceiver receiver,
                          Trackers trackers,
                          Subscription subscription,
                          Topic topic) {
-        this.receiver = new MessageBatchReceiver(receiver, batchFactory, hermesMetrics, messageConverterResolver, messageContentWrapper, topic, trackers);
+        this.messageReceiverFactory = messageReceiverFactory;
         this.sender = sender;
         this.batchFactory = batchFactory;
         this.offsets = offsets;
         this.subscription = subscription;
         this.hermesMetrics = hermesMetrics;
         this.monitoring = new BatchMonitoring(hermesMetrics, trackers);
+        this.messageConverterResolver = messageConverterResolver;
+        this.messageContentWrapper = messageContentWrapper;
+        this.topic = topic;
+        this.trackers = trackers;
     }
 
     @Override
     public void run() {
         setThreadName();
+        receiver = initializeMessageReceiver();
```

Comment:
Actually we could log on `info` how long bootstrap took and on `debug` what was going on...